chainerrl/agents/ppo.py

```diff
@@ -458,19 +776,41 @@ def batch_observe_and_train(self, batch_obs, batch_reward,
         )):
             if state is not None:
                 assert action is not None
-                self.batch_last_episode[i].append({
+                transition = {
                     'state': state,
                     'action': action,
                     'reward': reward,
                     'next_state': next_state,
                     'nonterminal': 0.0 if done else 1.0,
-                })
+                }
+                if self.recurrent:
+                    transition['recurrent_state'] =\
+                        self.model.get_recurrent_state_at(
+                            self.train_prev_recurrent_states,
+                            i, unwrap_variable=True)
+                    transition['next_recurrent_state'] =\
+                        self.model.get_recurrent_state_at(
+                            self.train_recurrent_states,
+                            i, unwrap_variable=True)
+                self.batch_last_episode[i].append(transition)
             if done or reset:
                 assert self.batch_last_episode[i]
                 self.memory.append(self.batch_last_episode[i])
                 self.batch_last_episode[i] = []
-                self.batch_last_state[i] = None
-                self.batch_last_action[i] = None
+            self.batch_last_state[i] = None
+            self.batch_last_action[i] = None
```

Comment:
Is the change to the 2 lines relevant to the recurrent PPO?

Reply:
I think it is just for clarifying that these values are no longer used, helping debugging.